src/bp/ui-admin/src/Pages/Workspace/Bots/CreateBotModal.tsx

```diff
@@ -139,9 +142,13 @@ class CreateBotModal extends Component<Props, State> {
     this.props.toggle()
   }
 
+  isIdUnique() {
+    return !this.props.bots.some(e => e.id === this.state.botId)
```

Comment:
I think !!this.props.bots.find  would be more appropriate since some means "possibly more than one" but a ID is unique

Reply:
The only thing I don't like with your suggestion is that you are doing a conversion with the double negation. At the end you don't need the value (bot is returned by the find). "Some" is achieving what we want: verify that there is at least one element that passes the test implemented by the provided function. Some in JavaScript is for me the equivalent of "any" in c#.

Reply:
The only thing I don't like with your suggestion is that you are doing a conversion with the double negation. At the end you don't need the value (bot is returned by the find). "Some" is achieving what we want: verify that there is at least one element that passes the test implemented by the provided function. Some in JavaScript is for me the equivalent of "any" in c#.